### PR TITLE
Update AbstractScalarParam.php

### LINKED ISSUE #1496: allowBlank and nullable no longer work together as they did in 1.0
Previously you could specify allowBlank=false and nullable=true, which would account for allowing the user to not specify an API parameter, but if they DO specify the parameter, that parameter cannot be blank.

Example:

```
* @RequestParam(name="token", requirements=".{1,200}", allowBlank=false, strict=true, nullable=true, description="User's token")
```

This worked correctly in 1.0 where `/api/call?token=` would throw an error, but `/api/call?` would be fine.

Now in 2.0, however, the above is throwing an exception when you do not specify the parameter at all.

"Parameter \"token\" of value \"\" violated a constraint \"This value should not be blank.\""

I think this is incorrect, because nullable was set to true, so the value should be allowed to be null. The parameter is incorrectly being flagged as blank, but it is in fact null since it was not included at all in the request.

### PR DESCRIPTION
**Pull request by Gonzalo1987**
Fixed #1496

I don't know if there is neccesary make any test... I'm not a usual contributor to this project, but I believe that this fix solve the compatibility problem between nullable=true and allowBlank=false 

Greetings!

### CODE REVIEW

File: Controller/Annotations/AbstractScalarParam.php [hunk 1]
```diff
@@ -65,7 +65,9 @@ public function getConstraints()
         }
 
         if (false === $this->allowBlank) {
-            $constraints[] = new NotBlank();
+            $notBlank = new NotBlank();
+            $notBlank->allowNull = $this->nullable;
```
Comment (xabbuh) [inline Controller/Annotations/AbstractScalarParam.php]: We need to do this conditionally as the `allowNull` option didn't exist before Symfony 4.3.

Comment (Gonzalo1987) [reply Controller/Annotations/AbstractScalarParam.php]: I have added the isset($notBlank->allowNull), thanks!

File: Controller/Annotations/AbstractScalarParam.php [hunk 2]
```diff
@@ -65,7 +65,11 @@ public function getConstraints()
         }
 
         if (false === $this->allowBlank) {
-            $constraints[] = new NotBlank();
+            $notBlank = new NotBlank();
+            if (isset($notBlank->allowNull)) {
```
Comment (xabbuh) [inline Controller/Annotations/AbstractScalarParam.php]: Are you sure this works as expected? I would expect this to always evaluate to `false` (see also https://3v4l.org/sdhnT). Probably using `property_exists()` is what we need to use here.

Comment (Gonzalo1987) [reply Controller/Annotations/AbstractScalarParam.php]: For test it, I have commented the $allowNull value of the Constraint class and I didn't get any new error, It worked like expected (still having the bug #1496, but not adding new exceptions). I can use the property_exists if it is cleaner so.

Comment (xabbuh) [reply Controller/Annotations/AbstractScalarParam.php]: If we didn't use `property_exists()`, our code would depend on the default value of the `$allowNull` property of the constraint class.

Comment (Gonzalo1987) [reply Controller/Annotations/AbstractScalarParam.php]: Thank you very much for your recomendation @xabbuh, I've modified the condition according your advise. I believe that now it's fine.

**Comment by Gonzalo1987** [comment]
@xabbuh It's now ok?

**Comment by Gonzalo1987** [comment]
Why this PR remains as 'Changes requested'?

**Comment by GuilhemN** [comment]
Thank you @Gonzalo1987! :)

**Pull request merged**